src/main/java/mate/academy/spring/security/CustomUserDetailsServiceImp.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.spring.security;
+
+import java.util.NoSuchElementException;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsServiceImp implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsServiceImp(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userService.findByEmail(username)
+                .orElseThrow(() -> new NoSuchElementException("UsernameNotFoundException"));
+        org.springframework.security.core.userdetails.User.UserBuilder builder =
+                org.springframework.security.core.userdetails.User.withUsername(username);
+        builder.password(user.getPassword());
+        builder.authorities(user.getRoles()
```

Comment:
```suggestion
        builder.roles(user.getRoles()
```
Как всегда, самые мелкие ошибки, забирают больше всего времени :upside_down_face:
Мы в конфиге проверяем роли и засечивать тоже должны били роли